keyboards/crkbd/keymaps/crkbd-test/keymap.c

```diff
@@ -0,0 +1,206 @@
+                  KC_LCTL,                KC_Z,                KC_X,                KC_C,                KC_V,                KC_B,                                                             KC_K,                KC_M,             KC_COMM,              KC_DOT,             KC_SLSH, S(LWIN(SWIN(KC_S))),
+  //|--------------------+--------------------+--------------------+--------------------+--------------------+--------------------+--------------------|  |--------------------+--------------------+--------------------+--------------------+--------------------+--------------------+--------------------|
+                                                                                                      KC_DEL,              KC_BSPC,               KC_UP,                 KC_ENT,              KC_SPC,               TG(6)
+                                                                                      //`--------------------------------------------------------------'  `--------------------------------------------------------------'
+  )
+};
+
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  if (!is_keyboard_master()) {
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  }
+  return rotation;
+}
+
+#define L_BASE 0
+#define L_LOWER 2
+#define L_RAISE 4
+#define L_ADJUST 8
+
+void oled_render_layer_state(void) {
+    oled_write_P(PSTR("Layer: "), false);
+    switch (layer_state) {
+        case L_BASE:
+            oled_write_ln_P(PSTR("Default"), false);
+            break;
+        case L_LOWER:
+            oled_write_ln_P(PSTR("Lower"), false);
+            break;
+        case L_RAISE:
+            oled_write_ln_P(PSTR("Raise"), false);
+            break;
+        case L_ADJUST:
+        case L_ADJUST|L_LOWER:
+        case L_ADJUST|L_RAISE:
+        case L_ADJUST|L_LOWER|L_RAISE:
+            oled_write_ln_P(PSTR("Adjust"), false);
+            break;
+    }
+}
+
+
+char keylog_str[24] = {};
+
+const char code_to_name[60] = {
+    ' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+    'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+    '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+    'R', 'E', 'B', 'T', '_', '-', '=', '[', ']', '\\',
+    '#', ';', '\'', '`', ',', '.', '/', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",      record->event.key.row, record->event.key.col,      keycode, name);
```

Comment:
sprintf usage should be avoided on AVR, as it adds a non-trival amount of size to compiled firmware (even with LTO enabled). 

eg:
https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

Reply:
Hi, thank you for the suggestions. This was originally meant for personal use and I made a mistake when doing the git stuff. Because of that, may I close this?

Reply:
No worries! 

And yeah, git ... can be a PITA at first!